numba/npyufunc/ufuncbuilder.py

```diff
@@ -264,7 +264,11 @@ def __init__(self, py_func, signature, identity=None, cache=False,
         self.identity = parse_identity(identity)
         self.nb_func = jit(target='npyufunc', cache=cache)(py_func)
         self.signature = signature
-        self.sin, self.sout = parse_signature(signature)
+        # allow internal use to pass in data structure that represents signature
+        if isinstance(signature, tuple):
```

Comment:
You are right. Both the above and this one are fixed in commit cab52c66852de8780bb63a739de1dce59200ab19. Thanks for spotting them!